relay-server/src/actors/events.rs

```diff
@@ -1133,6 +1139,22 @@ impl EventProcessor {
         Ok(())
     }
 
+    /// Run dynamic sampling rules to see if we keep the event or remove it.
+    fn sample_event(&self, state: &mut ProcessEnvelopeState) -> Result<(), ProcessingError> {
+        let event = match &state.event.0 {
+            None => return Ok(()), // can't process without an event
+            Some(event) => event,
+        };
+
+        let project_state = state.project_state.deref();
+        let project_id = state.project_id;
+        match utils::should_keep_event(event, project_state, project_id) {
```

Comment:
You should be able to directly pass `&state.project_state` in here and could get rid of the `Deref` import. Can you check?